addons/analytic/static/src/js/analytic_distribution/analytic_distribution.js

```diff
@@ -0,0 +1,606 @@
+        if (this.editingRecord && !this.isDropdownOpen) {
+            this.openAnalyticEditor();
+        }
+        if (this.isDropdownOpen) {
+            this.setFocusSelector('.tag_' + id + ' .o_analytic_percentage');
+            this.focusToSelector();
+            ev.stopPropagation();
+        }
+    }
+
+    // Focus
+    onMainElementFocus(ev) {
+        if (!this.isDropdownOpen && !this.recentlyClosed) {
+            this.openAnalyticEditor();
+        }
+    }
+
+    focusToSelector() {
+        if (!!this.focusSelector && this.isDropdownOpen) {
+            this.focus(this.adjacentElementToFocus("next", this.dropdownRef.el.querySelector(this.focusSelector)));
+        }
+        this.focusSelector = false;
+    }
+
+    setFocusSelector(selector) {
+        this.focusSelector = selector;
+    }
+
+    adjacentElementToFocus(direction, el = null) {
+        if (!this.isDropdownOpen) {
+            return null;
+        }
+        if (!el) {
+            el = this.dropdownRef.el;
+        }
+        return direction == "next" ? getNextTabableElement(el) : getPreviousTabableElement(el);
+    }
+
+    focusAdjacent(direction) {
+        let elementToFocus = this.adjacentElementToFocus(direction);
+        if (elementToFocus){
+            this.focus(elementToFocus);
+            return true;
+        }
+        return false;
+    }
+
+    focus(el) {
+        if (!el) return;
+        console.log('focusing to', el);
+        el.focus();
+        if (["INPUT", "TEXTAREA"].includes(el.tagName)) {
+            if (el.selectionStart) {
+                //bad
+                el.selectionStart = 0;
+                el.selectionEnd = el.value.length;
+            }
+            el.select();
+        }
+    }
+
+    // Keys and Clicks
+    onWidgetKeydown(ev) {
+        if (!this.editingRecord) {
+            return;
+        }
+        const hotkey = getActiveHotkey(ev);
+        switch (hotkey) {
+            case "enter":
+            case "tab": {
+                if (this.isDropdownOpen) {
+                    if (this.focusAdjacent("next")){
+                        break;
+                    } else {
+                        this.closeAnalyticEditor();
+                    }
+                };
+                this.resetRecentlyClosed();
+                return;
+            }
+            case "shift+tab": {
+                if (this.isDropdownOpen) {
+                    if (this.focusAdjacent("previous")){
+                        break;
+                    } else {
+                        this.closeAnalyticEditor();
+                        break;
+                    }
+                };
+                this.resetRecentlyClosed();
+                return;
+            }
+            case "escape": {
+                if (this.isDropdownOpen) {
+                    this.forceCloseEditor();
+                    break;
+                }
+                this.resetRecentlyClosed();
+                return;
```

Comment:
So basically, you always do this in all the cases
```js
                if (this.isDropdownOpen) {
                    ...
                }
                this.resetRecentlyClosed();
                return;
```

So why is it not done before the switch/case?

And don't you want to prevent the defaults/propagation in this case too?

Reply:
no, only for these keypresses - but not all